src/main/java/org/testng/internal/Parameters.java

```diff
@@ -778,6 +778,9 @@ public static ParameterHolder handleParameters(
     ParameterOrigin origin;
 
     if (null != dataProviderMethod) {
+      if (testMethod instanceof TestNGMethod) {
```

Comment:
In fact, we should change the param type but I suppose it will have too much impacts